src/io/inputs/model-converter/include/antares/io/inputs/model-converter/convertorVisitor.h

```diff
@@ -32,6 +38,37 @@ class NoPortWithThisId final: public std::runtime_error
     explicit NoPortWithThisId(const std::string& name);
 };
 
-Expressions::NodeRegistry convertExpressionToNode(const std::string& exprStr,
-                                                  const YmlModel::Model& model);
+Modeler::Config::Location convertLocation(const std::string& locationStr);
+
+template<class T>
+auto SearchEntity(const std::string id, std::vector<std::pair<T*, bool>>& values)
+{
+    return std::ranges::find_if(values,
+                                [&](const auto& optimConfigEntity)
+                                { return optimConfigEntity.first->id == id; });
+}
+
+template<class T>
+auto SearchEntity(const std::string id, const std::vector<std::pair<T*, bool>>& values)
+{
+    return std::ranges::find_if(values,
+                                [&](const auto& optimConfigEntity)
+                                { return optimConfigEntity.first->id == id; });
+}
+
+template<class T>
+Modeler::Config::Location updateLocation(std::string id, std::vector<std::pair<T*, bool>>& values)
```

Comment:
If you want to clarify the intent you can add a const
```suggestion
Modeler::Config::Location updateLocation(std::string id, std::vector<std::pair<const T*, bool>>& values)
```